dlt/destinations/dataset/relation.py

```diff
@@ -21,80 +24,159 @@
+        self.__normalized_query: sge.Query = None
+
+        # parse incoming query object
+        self._sqlglot_expression: sge.Query = None
+        if IbisExpr and isinstance(query, IbisExpr):
+            self._sqlglot_expression = compile_ibis_to_sqlglot(query, self.query_dialect())
+        elif query:
+            self._sqlglot_expression = maybe_parse(
+                query,
+                dialect=query_dialect or self.query_dialect(),
+            )
+        else:
+            self._sqlglot_expression = build_select_expr(
+                table_name=table_name,
+                selected_columns=list(self._dataset.schema.get_table_columns(table_name).keys()),
+            )
 
-        # wire protocol functions
-        self.df = self._wrap_func("df")  # type: ignore
-        self.arrow = self._wrap_func("arrow")  # type: ignore
-        self.fetchall = self._wrap_func("fetchall")  # type: ignore
-        self.fetchmany = self._wrap_func("fetchmany")  # type: ignore
-        self.fetchone = self._wrap_func("fetchone")  # type: ignore
+    #
+    # forward DataAccess protocol methods
+    #
+    def _wrap_iter(self, func_name: str) -> Any:
+        """wrap Relation generators in cursor context"""
 
-        self.iter_df = self._wrap_iter("iter_df")  # type: ignore
-        self.iter_arrow = self._wrap_iter("iter_arrow")  # type: ignore
-        self.iter_fetch = self._wrap_iter("iter_fetch")  # type: ignore
+        def _wrap(*args: Any, **kwargs: Any) -> Any:
+            with self.cursor() as cursor:
+                yield from getattr(cursor, func_name)(*args, **kwargs)
+
+        return _wrap
+
+    def _wrap_func(self, func_name: str) -> Any:
+        """wrap Relation functions in cursor context"""
+
+        def _wrap(*args: Any, **kwargs: Any) -> Any:
+            with self.cursor() as cursor:
+                return getattr(cursor, func_name)(*args, **kwargs)
+
+        return _wrap
+
+    def df(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_func("df")(*args, **kwargs)
+
+    def arrow(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_func("arrow")(*args, **kwargs)
+
+    def fetchall(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_func("fetchall")(*args, **kwargs)
+
+    def fetchmany(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_func("fetchmany")(*args, **kwargs)
+
+    def fetchone(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_func("fetchone")(*args, **kwargs)
+
+    def iter_df(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_iter("iter_df")(*args, **kwargs)
+
+    def iter_arrow(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_iter("iter_arrow")(*args, **kwargs)
+
+    def iter_fetch(self, *args: Any, **kwargs: Any) -> Any:
+        return self._wrap_iter("iter_fetch")(*args, **kwargs)
 
+    @property
+    def columns_schema(self) -> TTableSchemaColumns:
```

Comment:
hmmm maybe we can convert this into `schema`? and return `TTableSchema`? why? to propagate table hints as well. I'm pretty sure we'll have to change that soon

Reply:
I have converted it to a schema property now that also is on the protocol. I need to convert it back to a hints format though to be able to use it in the transform resource.